### PR TITLE
feat(graphql_parser): parse input object type extension

### PR DESCRIPTION
**Pull request by vohoanglong0107**
## Summary

Parse GraphQL input object extensions ([link](https://spec.graphql.org/October2021/#sec-Input-Object-Extensions)). We should be ready to benchmark the GraphQL parser after this.

## Test Plan

All tests should pass.

### CODE REVIEW

File: crates/biome_graphql_parser/tests/graphql_test_suite/err/definitions/input_object_extension.graphql.snap [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  × expected `{` but instead found `x`
+  
+    3 │ extend input Point2D
+  > 4 │   x: Float
+      │   ^
+    5 │   y: Float
+    6 │ }
+  
+  i Remove x
```
Comment (arendjr) [inline crates/biome_graphql_parser/tests/graphql_test_suite/err/definitions/input_object_extension.graphql.snap:149]: I feel this suggestion could be improved.

**Review by arendjr — APPROVED** [review]
👍 Just a minor comment.

File: crates/biome_graphql_parser/tests/graphql_test_suite/err/definitions/input_object_extension.graphql.snap (hunk 1 above)
Comment (vohoanglong0107) [reply crates/biome_graphql_parser/tests/graphql_test_suite/err/definitions/input_object_extension.graphql.snap:149]: I will create a new PR to address this, as the current PR does not modify this behavior but only copies the error message from the input object definition rule. What would you suggest for the new suggestion?

Comment (arendjr) [reply crates/biome_graphql_parser/tests/graphql_test_suite/err/definitions/input_object_extension.graphql.snap:149]: It seems the code could be fixed by adding the `{`, so maybe a suggestion along the lines of “did you forget to add the `{`”?

**Pull request merged**